jupyterhub/tests/test_api.py

```diff
@@ -27,7 +28,7 @@
 from .utils import auth_header
 from .utils import find_user
 
-
+ssl_enabled = os.environ.get('SSL_ENABLED', False)
```

Comment:
Instead of defining this on each module, can we load it centrally (i.e. *replace* the request.module.ssl_enabled) check with this env check? Or if we still want it sometimes, use something like `getattr(request.module, ssl_enabled, os.environ.get...)`

Reply:
That was my first attempt, but I ran into trouble since the tests in e.g. `test_proxy.py` error out due to failing SSL certificates. I thought the `ssl_enabled` module variable was best left in to be able to differentiate between tests that allow for internal SSL and tests that don't.

Alternatively, I can fix the tests that fail under internal SSL?

Reply:
That is worthwhile to investigate, I think! The other option would be to make it globally on by default and then skip the tests that require not-ssl if ssl is enabled.